packages/thirdweb/src/transaction/prepare-transaction.ts

```diff
@@ -25,10 +25,20 @@ export type StaticPrepareTransactionOptions = {
   client: ThirdwebClient;
   // extras
   extraCallData?: Hex;
-  erc20Value?: {
-    amountWei: bigint;
-    tokenAddress: Address;
-  };
+  erc20Value?:
+    | {
+        amountWei: bigint;
+        tokenAddress: Address;
+      }
+    | Promise<
```

Comment:
make sure we're handling this being a promise where its used

Reply:
so far I believe only getApprovalForTransaction is using it (which is the reason why you added erc20Value in the first place?) - I'll double check anyway